dubbo-registry/dubbo-registry-redis/src/main/java/org/apache/dubbo/registry/redis/RedisRegistry.java

```diff
@@ -86,42 +86,15 @@ public RedisRegistry(URL url) {
         if (url.isAnyHost()) {
             throw new IllegalStateException("registry address == null");
         }
-        GenericObjectPoolConfig config = new GenericObjectPoolConfig();
-        config.setTestOnBorrow(url.getParameter("test.on.borrow", true));
-        config.setTestOnReturn(url.getParameter("test.on.return", false));
-        config.setTestWhileIdle(url.getParameter("test.while.idle", false));
-        if (url.getParameter("max.idle", 0) > 0) {
-            config.setMaxIdle(url.getParameter("max.idle", 0));
-        }
-        if (url.getParameter("min.idle", 0) > 0) {
-            config.setMinIdle(url.getParameter("min.idle", 0));
-        }
-        if (url.getParameter("max.active", 0) > 0) {
-            config.setMaxTotal(url.getParameter("max.active", 0));
-        }
-        if (url.getParameter("max.total", 0) > 0) {
-            config.setMaxTotal(url.getParameter("max.total", 0));
-        }
-        if (url.getParameter("max.wait", url.getParameter("timeout", 0)) > 0) {
-            config.setMaxWaitMillis(url.getParameter("max.wait", url.getParameter("timeout", 0)));
-        }
-        if (url.getParameter("num.tests.per.eviction.run", 0) > 0) {
-            config.setNumTestsPerEvictionRun(url.getParameter("num.tests.per.eviction.run", 0));
-        }
-        if (url.getParameter("time.between.eviction.runs.millis", 0) > 0) {
-            config.setTimeBetweenEvictionRunsMillis(url.getParameter("time.between.eviction.runs.millis", 0));
-        }
-        if (url.getParameter("min.evictable.idle.time.millis", 0) > 0) {
-            config.setMinEvictableIdleTimeMillis(url.getParameter("min.evictable.idle.time.millis", 0));
-        }
+        GenericObjectPoolConfig config = new URLJedisConfigAdapter(url);
 
         String cluster = url.getParameter("cluster", "failover");
         if (!"failover".equals(cluster) && !"replicate".equals(cluster)) {

```

Comment:
Would you please replace the 'failover' with Constants.DEFAULT_CLUSTER?

Reply:
I think you can also put the `replicate` into the Constants.java.

Reply:
Sure, changes are made as requested